test/org/ojalgo/optimisation/convex/LagrangeTest.java

```diff
@@ -151,6 +151,30 @@ public void testNocedalAndWrightExample() throws RecoverableCondition {
         Optional<Access1D<?>> multipliers = result.getMultipliers();
         TestUtils.assertTrue("No multipliers present", multipliers.isPresent());
         TestUtils.assertEquals("Lagrangian Multipliers differ", expectedDual, multipliers.get(), accuracy);
+        
+//  Test similar system where each equality constraint are converted into two inequality constraints.
+//  The result should be the same.
+       Builder ieBuilder = ConvexSolver.getBuilder();
+       ieBuilder.objective(Q, C.negate());
+       MatrixStore<Double> AI = AE.logical().below(AE.negate()).get();
+       MatrixStore<Double> BI = BE.logical().below(BE.negate()).get();
+       ieBuilder.inequalities(AI, BI);
+       ConvexSolver ieSolver = ieBuilder.build();
+       
+       Result ieResult = ieSolver.solve();
+       Optional<Access1D<?>> ieMultipliers = ieResult.getMultipliers();
+       TestUtils.assertTrue("No multipliers present", ieMultipliers.isPresent());
+
+       int greaterThanCount = 0;
+       int greaterThanOrEqualCount = 0;
+       for (double v : ieMultipliers.get().toRawCopy1D()) {
+          if (v == 0)greaterThanOrEqualCount++;
+          if (v > 0) greaterThanCount++;
+       }
+// Test that two multipliers are greater than zero and two are zero.
+       TestUtils.assertTrue(greaterThanOrEqualCount == 2);
+       TestUtils.assertTrue(greaterThanCount == 2);
+       
```

Comment:
`greaterThanOrEqualCount` should perhaps be named `zeroCount`. Don't you know more precisely the values and positions? Should be safe test against a vector `{ 0.0 , 2.0, 3.0, 0.0 }`?

Reply:
Thanks for your comments. I modified the code to test first X and then the multipliers against fixed values as you suggested.